app.js

```diff
@@ -12,7 +14,13 @@ app.get('/', (req, res) => {
   res.send('This is a test response message');
 })
 
-app.get('/shipping-api/:productId')
+app.get('/shipping-api/:productId', (req, res) => {
+  mongoose.connect(process.env.DB_CONNECTION, {useNewUrlParser: true});
```

Comment:
I once had this experience. From line 18 to 20, I would recommend moving this code outside of the router, since you create a connection every time you call this endpoint.

Reply:
I'm going to write a bug for my backlog to investigate this more. 

If you think it will break things let me know, but I think this will work as is. I'll talk to you more about it too because I'm not quite sure I understand completely but am interested to see an alternate way to do this.

Reply:
AH I get it now!  Took a few hours to sink in but yes you are right - every time any request is made to the server it's going to open a connection to my db whether I need it or not. Excellent catch!  Sorry it took awhile to sink in!